src/Test.UnitTests.Sarif/Comparers/ComparersTests.cs

```diff
@@ -0,0 +1,802 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.CodeAnalysis.Sarif.Comparers;
+using Microsoft.CodeAnalysis.Test.Utilities.Sarif;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Comparers
+{
+    public class ComparersTests
+    {
+        private readonly Random random;
+
+        public ComparersTests(ITestOutputHelper outputHelper)
+        {
+            this.random = RandomSarifLogGenerator.GenerateRandomAndLog(outputHelper);
+        }
+
+        [Fact]
+        public void CompareList__Shuffle_Tests()
```

Comment:
>CompareList__Shuffle_Tests [](http://example.com/codeflow?start=20&length=26)

Go make this test fail. Show us the test output that makes it easy for a developer to repro the failure, using the random seed that generated the failing case.

Reply:
Rand created with RandomSarifLogGenerator.GenerateRandomAndLog logs the seed for the Random object it created in test output. Below is an example:
```
[xUnit.net 00:00:25.02]     Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Comparers.ComparersTests.CompareList_Shuffle_Tests [FAIL]
  Failed Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Comparers.ComparersTests.CompareList_Shuffle_Tests [203 ms]
  Error Message:
   Expected result to be 0, but found -1.
  Stack Trace:
     at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   at FluentAssertions.Execution.DefaultAssertionStrategy.HandleFailure(String message)
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
   at FluentAssertions.Numeric.NumericAssertions`1.Be(T expected, String because, Object[] becauseArgs)
   at Microsoft.CodeAnalysis.Test.UnitTests.Sarif.Comparers.ComparersTests.CompareList_Shuffle_Tests() in C:\repo\github.com\sarif-sdk\src\Test.UnitTests.Sarif\Comparers\ComparersTests.cs:line 40
  Standard Output Messages:
 TestName: CompareList_Shuffle_Tests has seed 727824218
```
Also updated the RandomSarifLogGenerator.GenerateRandomAndLog to accept a specific seed so dev can easily reuse the seed in the failed case: 
`Random random = RandomSarifLogGenerator.GenerateRandomAndLog(this.output, seed: 727824218);`